From 8b9b250dfb944998e7219f8e83c5219c29e7aff5 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 20 May 2022 04:15:59 +0300
Subject: [PATCH] [SW-82717] rom: update recovery rr config

As per the update in gaudi3 security specification, only
secured firmware update is supported in a production system.
Hence rr config in recovery is updated to only open registers
required in secured firmware update.

Change-Id: I84e87c4ba9d0179e2a2c0554e8d37ff39fc041fc
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index e8d9c9e..327a213 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -517,10 +517,10 @@
 	security_rr_twi_lbw_config(&rr_ctx_lbw[RR_OPEN_ALL]);
 
 	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_0;
-	security_rr_twi_lbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
+	security_rr_twi_hbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
 
 	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_7;
-	security_rr_twi_lbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
+	security_rr_twi_hbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
 }
 
 void security_disable_twi_if(void)
@@ -535,10 +535,10 @@
 	security_rr_twi_lbw_config(&rr_ctx_lbw[RR_BLOCK_ALL]);
 
 	rr_ctx_hbw[RR_BLOCK_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_0;
-	security_rr_twi_lbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
+	security_rr_twi_hbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
 
 	rr_ctx_hbw[RR_BLOCK_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_7;
-	security_rr_twi_lbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
+	security_rr_twi_hbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
 }
 
 void security_handle_twi(void)
@@ -791,42 +791,24 @@
 static void security_rr_recovery_lbw_config(void)
 {
 	/*
-	 * PSOC_UART_0 base to PSOC_GLOBAL_CONF start
-	 * 0x0300007FFC485000 - 0x0300007FFC48B000
+	 * PSOC_BTL to PSOC_SECURITY start
+	 * 0300007FFC48C000 - 0300007FFC494000
 	 */
 	struct rom_rr_ctx rr_ctx = {
 			.rr_idx = HL_SEC_LBW_RR_SHORT_1,
-			.data_min = (uint32_t)mmD0_PSOC_UART_0_BASE,
-			.data_max = (uint32_t)mmD0_PSOC_GLOBAL_CONF_BASE,
+			.data_min = (uint32_t)mmD0_PSOC_BTL_BASE,
+			.data_max = (uint32_t)mmD0_PSOC_SECURITY_BASE,
 			.en_val = RR_PRIV_RANGE_EN,
 	};
 
 	security_rr_config_lbw_if_common(&rr_ctx);
 
 	/*
-	 * PSOC_BTL base to PSOC Security start
-	 * 0x0300007FFC48C000 - 0x0300007FFC494000
-	 */
-	rr_ctx.rr_idx = HL_SEC_LBW_RR_SHORT_2;
-	rr_ctx.data_min = (uint32_t)mmD0_PSOC_BTL_BASE;
-	rr_ctx.data_max = (uint32_t)mmD0_PSOC_SECURITY_BASE;
-	security_rr_config_lbw_if_common(&rr_ctx);
-
-	/*
-	 * JT_MSTR_IF start to PSOC_GPIO_DATAOUT
-	 * 0x0300007FFC495000 - 0x300007FFC4C0004
+	 * JT_MSTR_IF start to mmD0_PARC_ARC1_CFG_KMD_HW_DIRTY_STATUS
+	 * 0300007FFC495000 - 0x0300007FFC4E3200
 	 */
 	rr_ctx.rr_idx = HL_SEC_LBW_RR_FULL_0;
 	rr_ctx.data_min = (uint32_t)mmD0_JT_MSTR_IF_RR_HBW_BASE;
-	rr_ctx.data_max = HL_CFG_BASE + mmD0_PSOC_GPIO0_DATAOUT;
-	security_rr_config_lbw_if_common(&rr_ctx);
-
-	/*
-	 * PSOC_GPIO_OUTENCLR base to PARC_ARC1_KMD_HW_DIRTY_STATUS
-	 * 0x0300007FFC48C000 - 0x0300007FFC4E3200
-	 */
-	rr_ctx.rr_idx = HL_SEC_LBW_RR_FULL_1;
-	rr_ctx.data_min = HL_CFG_BASE + mmD0_PSOC_GPIO0_OUTENCLR;
 	rr_ctx.data_max = HL_CFG_BASE + mmD0_PARC_ARC1_CFG_KMD_HW_DIRTY_STATUS;
 	security_rr_config_lbw_if_common(&rr_ctx);
 
@@ -834,18 +816,18 @@
 	 * ARC1_CFG_PF_PQ_PI base to CFG_END
 	 * 0x0300007FFC4E3204 - 0x300007FFFFFF000
 	 */
-	rr_ctx.rr_idx = HL_SEC_LBW_RR_FULL_2;
-	rr_ctx.data_min =  HL_CFG_BASE + mmD0_PARC_ARC1_CFG_PF_PQ_PI;
-	rr_ctx.data_max = RR_LBW_PRIV_RANGE_MAX_SHORT_VAL;
+	rr_ctx.rr_idx = HL_SEC_LBW_RR_FULL_1;
+	rr_ctx.data_min = HL_CFG_BASE + mmD0_PARC_ARC1_CFG_PF_PQ_PI;
+	rr_ctx.data_max = RR_LBW_PRIV_RANGE_MAX_FULL_VAL;
 	security_rr_config_lbw_if_common(&rr_ctx);
 
 	/*
-	 * CFG_SPACE base to PSOC_QSPI
-	 * 0x0300007FE0000000 - 0x0300007FFC484000
+	 * CFG_SPACE base to PSOC_GLOBAL_CONF base
+	 * 0x0300007FE0000000 - 0x0300007FFC48B000
 	 */
 	rr_ctx.rr_idx = HL_SEC_LBW_RR_SHORT_0;
 	rr_ctx.data_min = RR_LBW_PRIV_RANGE_MIN_SHORT_VAL;
-	rr_ctx.data_max = (uint32_t)mmD0_PSOC_QSPI_BASE;
+	rr_ctx.data_max = (uint32_t)mmD0_PSOC_GLOBAL_CONF_BASE;
 	security_rr_config_lbw_if_common(&rr_ctx);
 
 	/* RR config by HW is neutralized */
